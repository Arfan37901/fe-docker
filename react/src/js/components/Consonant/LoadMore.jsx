import React from 'react';
import PropTypes from 'prop-types';

const LoadMore = (props) => {
    const { onClick, shown, total } = props;
    const render =
        (shown > 0 && total > 0) ? (
            <div className="consonant-load-more">
                <div className="consonant-load-more--inner">
                    <p className="consonant-load-more--text">
                        <span className="consonant-load-more--shown">{shown} </span>
                    of
                        <span className="consonant-load-more--total"> {total} </span>
                    displayed
                    </p>
                    {shown < total &&
                    <button
                        type="button"
                        className="consonant-load-more--btn"
                        onClick={onClick}>Load more
                    </button>
                    }
                </div>
            </div>)
            : null;
    return render;
};

export default LoadMore;

LoadMore.propTypes = {
    shown: PropTypes.number.isRequired,
    total: PropTypes.number.isRequired,
    onClick: PropTypes.func.isRequired,
};
