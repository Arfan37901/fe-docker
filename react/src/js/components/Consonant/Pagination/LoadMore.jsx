import React from 'react';
import { number, func } from 'prop-types';

import { useConfig } from '../Helpers/hooks';

const TLoadMore = {
    show: number.isRequired,
    total: number.isRequired,
    onClick: func.isRequired,
};

/**
 * Load More - Button That Naviates Users To The Next Page
 *
 * @component
 * @example
 * const props= {
    show: Int,
    total: Int,
    onClick: Function,
 * }
 * return (
 *   <LoadMore {...props}/>
 * )
 */
const LoadMore = ({
    show,
    total,
    onClick,
}) => {
    const getConfig = useConfig();

    /**
     * Authored Button Text
     * @type {String}
     */
    const loadMoreButtonText = getConfig('pagination', 'i18n.loadMore.btnText');

    /**
     * Authored Summary Text
     * @type {String}
     */
    const loadMoreQuantityText = getConfig('pagination', 'i18n.loadMore.resultsQuantityText');

    /**
     * Summary Of Load More Results To Show To Users
     * @type {String}
     */
    const summaryText = loadMoreQuantityText
        .replace('{start}', show)
        .replace('{end}', total);

    const shouldDisplayLoadMore = show > 0 && total > 0;
    const shouldDisplayLoadMoreBtn = show < total;

    return (shouldDisplayLoadMore) ? (
        <div
            data-testid="consonant-load-more"
            className="consonant-load-more">
            <div className="consonant-load-more--inner">
                <p
                    data-testid="consonant-load-more--text"
                    className="consonant-load-more--text">
                    {summaryText}
                </p>
                {shouldDisplayLoadMoreBtn &&
                    <button
                        type="button"
                        data-testid="load-more__button"
                        className="consonant-load-more--btn"
                        onClick={onClick}
                        tabIndex="0">
                        {loadMoreButtonText}
                    </button>
                }
            </div>
        </div>)
        : null;
};

LoadMore.propTypes = TLoadMore;

export default LoadMore;
