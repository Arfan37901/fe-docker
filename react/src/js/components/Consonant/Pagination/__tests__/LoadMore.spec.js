import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import LoadMore from '../LoadMore';

import { DEFAULT_PROPS } from '../../Testing/Constants/LoadMore';

import makeSetup from '../../Testing/Utils/Settings';

const setup = makeSetup(LoadMore, DEFAULT_PROPS);

describe('Consonant/LoadMore', () => {
    test('shouldn`t render if total is 0', () => {
        const { wrapper: { container } } = setup({ total: 0 });

        expect(container).toBeEmptyDOMElement();
    });
    test('shouldn`t render if show is 0', () => {
        const { wrapper: { container } } = setup({ show: 0 });

        expect(container).toBeEmptyDOMElement();
    });

    describe('Interaction with UI', () => {
        test('should call onClick', () => {
            const { props: { onClick } } = setup();

            const buttonElement = screen.getByTestId('load-more__button');

            fireEvent.click(buttonElement);

            expect(onClick).toBeCalled();
        });
    });
});
