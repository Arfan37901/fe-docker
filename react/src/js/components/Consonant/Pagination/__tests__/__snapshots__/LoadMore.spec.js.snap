// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/LoadMore Check snapshots should renders correctly with show < total 1`] = `
<div
  className="consonant-load-more"
  data-testid="consonant-load-more"
>
  <div
    className="consonant-load-more--inner"
  >
    <p
      className="consonant-load-more--text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span className=\\"consonant-load-more--shown\\">1</span> of <span className=\\"consonant-load-more--total\\">100</span> displayed",
        }
      }
      data-testid="consonant-load-more--text"
    />
    <button
      className="consonant-load-more--btn"
      onClick={[MockFunction]}
      tabIndex="0"
      type="button"
    >
      Load more
    </button>
  </div>
</div>
`;

exports[`Consonant/LoadMore Check snapshots should renders correctly with show > total 1`] = `
<div
  className="consonant-load-more"
  data-testid="consonant-load-more"
>
  <div
    className="consonant-load-more--inner"
  >
    <p
      className="consonant-load-more--text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<span className=\\"consonant-load-more--shown\\">100</span> of <span className=\\"consonant-load-more--total\\">1</span> displayed",
        }
      }
      data-testid="consonant-load-more--text"
    />
  </div>
</div>
`;
